### PR TITLE
Stop to write zero velocities when there is no input and set component to port_driven

### PR DESCRIPTION
**Pull request by dbcesar**
Stop to write zero velocities when there is no input and set component to port_driven

### CODE REVIEW

File: tasks/Task.cpp [hunk 1]
```diff
@@ -93,13 +93,17 @@ bool Task::configureHook()
     cVb.buildFrom(cMb);
     task.set_cVe(cVb);
 
+    //set the time for trigger the timeout
+    timeout = base::Timeout(base::Time::fromSeconds(1));
```
Comment (doudou) [inline tasks/Task.cpp]: Doesn't every single component in the control chain already has this behaviour. Including the velocity controller ?

Does it really make sense to add the same boilerplate complexity to all the control-related components ?

File: test/test.rb [hunk 2]
```diff
@@ -196,7 +196,7 @@ def create_corners(p1, p2, p3, p4)
 
     marker_corners.write corners_samples
     cmd_in.write setpoint_samples
-    sleep(0.01)
+    sleep(0.1)
```
Comment (doudou) [inline test/test.rb]: The sleeps make no sense here. If you're waiting for the new sample, the assert_has_one_new_sample is already waiting. Maybe need to set the assert's timeout (last argument) higher ?

File: tasks/Task.cpp (hunk 1 above)
Comment (dbcesar) [reply tasks/Task.cpp]: I added it in order to send a vector of zeros if the timeout is elapsed. 
Without this, the velocity controller goes to timeout so the robot tends to float. In that case, the expected behavior is keep the vehicle in the position if no markers are detected.

File: test/test.rb (hunk 2 above)
Comment (dbcesar) [reply test/test.rb]: Indeed. I edited this commit.

File: tasks/Task.cpp (hunk 1 above)
Comment (doudou) [reply tasks/Task.cpp]: > the expected behavior is keep the vehicle in the position if no markers are detected.

I don't agree. If we want this, then the velocity/position controllers should start controlling at zero if it they have no commands.

If you have no feedback, you have no business controlling anything.

Comment (dbcesar) [reply tasks/Task.cpp]: > I don't agree. If we want this, then the velocity/position controllers should start controlling at zero if it they have no commands.

Good point. 

Writing zero velocities aimed to handle two situations: 1) Eventual marker losing. 2) Big detection time. For (1) we should handle it on syskit level. For (2), we can just consider that if the feedback is not fast enough for control the process, then it is a not controllable system.

Considering this, at least for visual based control we should think about to increase the `aligned_vel2body_effort` timeout that currently it is 1 second. 
Apriltags detector running at 2K resolution detects the marker at 500ms average, but some times it goes to 1.2 seconds which would raise a TIMEOUT exception. I down-scale the image in order to achieve 100ms average, but experiences with SURF detectors can easily achieve 1s.

Moreover, what do you think about make this component `port_driven` ?

Comment (doudou) [reply tasks/Task.cpp]: is TIMEOUT an exception on the controllers ? I thought it was a runtime error...

I think it makes sense to have this component port driven on the image.

Comment (dbcesar) [reply tasks/Task.cpp]: Just asked @joaobrittoneto. 

I'll double check it.

Comment (dbcesar) [reply tasks/Task.cpp]: It is `error_state`. Sorry for that.

File: test/test.rb (hunk 2 above)
Comment (doudou) [reply test/test.rb]: The sleeps are still there....

File: visp.orogen [hunk 3]
```diff
@@ -47,7 +47,7 @@ needs_configuration
     output_port "controller_state", "visp/controllerState"
 
     #update the task every 10ms
-    periodic 0.01
+    periodic 0.1
```
Comment (doudou) [inline visp.orogen]: Weird. We just discussed to make the component port_driven on the image ...

**Comment by dbcesar** [comment]
@doudou  May I merge this?

**Comment by doudou** [comment]
If it passes the cucumber tests, yes.

**Comment by doudou** [comment]
Correcting myself: the current version is utterly broken and does not pass the tests itself ...

**Pull request merged**